test/regress_util.c

```diff
@@ -1536,6 +1540,169 @@ test_evutil_v6addr_is_local(void *arg)
 	;
 }
 
+#define SOCKETPAIR_CLOSE(fd)	do {	\
+	if (fd[0] != -1) evutil_closesocket(fd[0]);	\
+	if (fd[1] != -1) evutil_closesocket(fd[1]);	\
+} while(0)
+#define SOCKETPAIR_INIT(fd)		do {	\
+	fd[0] = -1; fd[1] = -1;				\
+} while(0)
+#define SOCKETPAIR_CHECK_CLOSE(fd)	do {\
+	tt_int_op(fd[0], > , 0);			\
+	tt_int_op(fd[1], > , 0);			\
+	SOCKETPAIR_CLOSE(fd);				\
```

Comment:
but this, should be a define, since it
- uses `tt_int_op`, and macro will preserve original `__file__`/`__line__`
- and also `tt_int_op` uses `goto end;`
So agree with this bit.